controllers/drpolicy_controller.go

```diff
@@ -55,8 +58,42 @@ func (r *DRPolicyReconciler) SetupWithManager(mgr ctrl.Manager) error {
 
 	dpPredicate := utils.ComposePredicates(predicate.GenerationChangedPredicate{})
 
+	tokenToDRPolicyMapFunc := func(ctx context.Context, obj client.Object) []ctrl.Request {
+		reqs := []ctrl.Request{}
+		var mirrorPeerList multiclusterv1alpha1.MirrorPeerList
+		err := r.HubClient.List(ctx, &mirrorPeerList)
+		if err != nil {
+			r.Logger.Error("Unable to reconcile DRPolicy based on token changes. Failed to list MirrorPeers.", "error", err)
+			return reqs
+		}
+		for _, mirrorpeer := range mirrorPeerList.Items {
+			for _, peerRef := range mirrorpeer.Spec.Items {
+				name := utils.CreateUniqueSecretName(peerRef.ClusterName, peerRef.StorageClusterRef.Namespace, peerRef.StorageClusterRef.Name)
+				if name == obj.GetName() {
+					var drpolicyList ramenv1alpha1.DRPolicyList
+					err := r.HubClient.List(ctx, &drpolicyList)
+					if err != nil {
+						r.Logger.Error("Unable to reconcile DRPolicy based on token changes. Failed to list DRPolicys", "error", err)
+						return reqs
+					}
+					for _, drpolicy := range drpolicyList.Items {
+						for i := range drpolicy.Spec.DRClusters {
+							if drpolicy.Spec.DRClusters[i] == peerRef.ClusterName {
+								reqs = append(reqs, reconcile.Request{NamespacedName: types.NamespacedName{Name: drpolicy.Name}})
+								break
```

Comment:
Is breaking early appropriate ? What if there are more DRPolicies having the same cluster name